From d728a6ed0471edbe19b79026d65cafa9c777f5d4 Mon Sep 17 00:00:00 2001
From: Stijn Tintel <stijn@linux-ipv6.be>
Date: Fri, 20 Aug 2021 16:11:12 +0300
Subject: [PATCH 014/149] uhttpd: add config option for json_script

Add a config option for json_script instead of unconditionally including
all json files in /etc/uhttpd in every uhttpd instance. This makes it
possible to configure a single instance with an unconditional redirect,
which is currently not possible as it would render all other uhttpd
instances unusable.

Signed-off-by: Stijn Tintel <stijn@linux-ipv6.be>
---
 package/network/services/uhttpd/files/uhttpd.init | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/package/network/services/uhttpd/files/uhttpd.init b/package/network/services/uhttpd/files/uhttpd.init
index e7709941c2..30fd7b4259 100755
--- a/package/network/services/uhttpd/files/uhttpd.init
+++ b/package/network/services/uhttpd/files/uhttpd.init
@@ -196,7 +196,8 @@ start_instance()
 		append_bool "$cfg" redirect_https "-q" 0
 	}
 
-	for file in /etc/uhttpd/*.json; do
+	config_get json_script "$cfg" json_script
+	for file in $json_script; do
 		[ -s "$file" ] && procd_append_param command -H "$file"
 	done
 
-- 
2.34.1

